src/main/java/ch/vorburger/exec/ManagedProcessBuilder.java

```diff
@@ -138,13 +138,13 @@ protected ManagedProcessBuilder addArgument(String argPart1, String separator, S
         final StringBuilder sb = new StringBuilder();
         final String arg;
 
-        // @see MariaDB4j Issue #501 Fix for spaces in data path doesn't work on windows
-        // https://github.com/vorburger/MariaDB4j/issues/501
-        // Internally Runtime.exec is being used, which says that an argument such
+        // @see https://github.com/vorburger/MariaDB4j/issues/501 - Fix for spaces in data path doesn't work on windows:
+        // Internally Runtime.exec() is being used, which says that an argument such
         // as --name="escaped value" isn't escaped, since there's no leading quote
-        // and it contains a space, so it reescapes it, causing applications to split
+        // and it contains a space, so it re-escapes it, causing applications such as mysqld.exe to split
         // it into multiple pieces, which is why we quote the whole arg (key and value) instead.
-        if ("=".equals(separator)) {
+        // We do this trick only on Windows, because on Linux it breaks the behavior.
+        if ("=".equals(separator) && isWindows()) {
```

Comment:
@Blanco27 I was quite tempted to further simplify this and ditch the `"=".equals(separator)` entirely here, but then opted to "be conservative" and propose it like this for you... but if I have a minute to try it out without that and just as `isWindows()` I guess that works as well for you on Windows, right? I won't really make any difference in MariaDB4j's usage anyway, but it seems simpler and (a bit, it's a mess anyway) cleaner, to me - what are your thoughts?

```suggestion
        if (isWindows()) {
```

Reply:
Tried it. The `isWindows()` is enough to make it work.

Reply:
OK then let's just use that; I'll change that and then merge this.